Frontend/Gmfctn/src/app/modules/+auth/services/authenticate.service.ts

```diff
@@ -0,0 +1,38 @@
+import { HttpClient, HttpErrorResponse } from '@angular/common/http';
+import { Injectable, OnDestroy } from '@angular/core';
+import { BehaviorSubject, Observable, of, Subject } from 'rxjs';
+import { catchError, take, tap } from 'rxjs/operators';
+
+import { Tokens } from '../../../shared/models/token';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthenticateService implements OnDestroy{
+  tokens$: BehaviorSubject<Tokens> = new BehaviorSubject(null as unknown as Tokens);
+  isErrorInAuthorization$: BehaviorSubject<boolean> = new BehaviorSubject(null as unknown as boolean);
+  private apiUrl = 'https://localhost:44349/';
+
+  constructor(private httpClient: HttpClient) { }
+
+  ngOnDestroy(): void {
+    this.tokens$.complete();
+    this.isErrorInAuthorization$.complete();
+  }
+
+  authenticate(Login: string, Password: string): Observable<string[]> {
+    return this.httpClient
+      .post<string[]>(`${this.apiUrl}api/auth/login?Login=${Login}&Password=${Password}`, '')
```

Comment:
Third parameter of the post method is options
There is a property called params, which you can use to pass the query params to the request as an object instead of combining it in the sting

Reply:
Yes it was my fault while I was constructing api. Now I have fixed api, and I can send this post with obj in body.

Reply:
Fixed